jupyter/start_jupyter.bash

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+cd /notebooks
+pip install pymor[full]
```

Comment:
This should also a write operation to the container @ashwin-nayak

Reply:
I only get a series of `Requirement already satisfied` messages to all the dependencies that needed to be installed with pip install pymor. This may change if any upstream dependencies complain, I suppose.

Reply:
Huh, the dependencies I understand, since those are intentionally baked into the image. But pyMOR itself is not, so where does pip find it?

Reply:
This is what `pip` says : 
`Requirement already satisfied: pymor[full] in /root/.local/lib/python3.9/site-packages (2022.1.1)`. 

I am not sure if you had installed it there during build.

Reply:
Could this be mounted from your "outside" filesystem?

Reply:
I sadly don't have `pymor` installed in my system and always have used it only through containers, so that is definitely not it. Also, not all system-directories are not mounted onto the containers by default. [More Info](https://apptainer.org/docs/user/main/bind_paths_and_mounts.html)

Reply:
I've checked to make sure it's not in the image:
```
docker run -it zivgitlab.wwu.io/pymor/docker/pymor/jupyter_py3.9:7173783179d446e051ee22ae14e5780d2eadd20b pip freeze | grep -i pymor
pymor-dealii @ file:///usr/local/tmp/pymor_dealii-2022.1.1.post0.dev10-cp39-cp39-linux_x86_64.whl
pymor-oldest-supported-numpy==2021.1.0
```

From your link, I think I now no why there's no error for the install. The first time you ran this image `pip install` tried to install into r/o `/usr/local`, failed, and fell back on `--user` install. Into what apptainer mounted has the full `$HOME` for the running container, which is writable (and for some reason `/root` and not /home/user?). That make sense to you?

Reply:
I verified `pymor` was for some reason on ${HOME}/.local/lib/python3.9/site-packages. I removed this and tried now with not sharing the home directory with `--no-home` option.
And as you guessed correctly, the runscript fails with,
`ERROR: Could not install packages due to an OSError: [Errno 30] Read-only file system: '/home/$USER'`

Reply:
Excellent. I think we've figured this out then and can merge this PR?

And for the future the takeaway to ensure apptainer compat would be to make system-level changes only at image build time, and entrypoints/commands should only need write access to home?

Reply:
Yup, definitely! I believe supporting `docker` is sufficient for `pymor` usecases for the moment. Thank you for taking this up.